borsh-derive-internal/src/enum_ser.rs

```diff
@@ -24,40 +26,55 @@ pub fn enum_ser(input: &ItemEnum, cratename: Ident) -> syn::Result<TokenStream2>
 
     let mut serialize_params_visitor = FindTyParams::new(&generics);
     let mut override_predicates = vec![];
+    let use_discriminant = contains_use_discriminant(input)?;
 
     let mut all_variants_idx_body = TokenStream2::new();
     let mut fields_body = TokenStream2::new();
     let discriminants = discriminant_map(&input.variants);
-    for variant in input.variants.iter() {
+
+    for (variant_idx, variant) in input.variants.iter().enumerate() {
+        let variant_idx = u8::try_from(variant_idx).expect("up to 256 enum variants are supported");
```

Comment:
```suggestion
        let variant_idx = u8::try_from(variant_idx).map_err(|err| {
            syn::Error::new(
                variant.ident.span(),
                format!("up to 256 enum variants are supported. error{}", err),
            )
        })?;
```
replace panic similarly to `enum_de.rs`